File: data/derived_cot/rq1_traces/translation/qwen/java_to_cpp/code_output/DataStatistics.cpp
#include <vector>
#include <algorithm>
#include <cmath>
#include <unordered_map>
#include <map>
#include <cstddef> // for size_t

class DataStatistics {
public:
    double mean(std::vector<int> data) {
        double sum = 0.0;
        for (int num : data) {
            sum += num;
        }
        double average = sum / static_cast<double>(data.size());
        return std::round(average * 100.0) / 100.0;
    }

    double median(std::vector<int> data) {
        std::vector<int> sortedData = data;
        std::sort(sortedData.begin(), sortedData.end());
        size_t n = sortedData.size();

        if (n % 2 == 0) {
            size_t middle = n / 2;
            double average = (sortedData[middle - 1] + sortedData[middle]) / 2.0;
            return std::round(average * 100.0) / 100.0;
        } else {
            size_t middle = n / 2;
            return sortedData[middle];
        }
    }

    std::vector<int> mode(std::vector<int> data) {
        std::unordered_map<int, int> frequencyMap;
        for (int num : data) {
            frequencyMap[num]++;
        }

        int maxCount = 0;
        for (const auto& pair : frequencyMap) {
            if (pair.second > maxCount) {
                maxCount = pair.second;
            }
        }

        std::vector<int> modes;
        for (const auto& pair : frequencyMap) {
            if (pair.second == maxCount) {
                modes.push_back(pair.first);
            }
        }

        std::sort(modes.begin(), modes.end());
        return modes;
    }
};

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/DataStatistics.py
import math

class NoSuchElementException(Exception):
    pass

class DataStatistics:

    def mean(self, data):
        n = len(data)
        if n == 0:
            return 0.0
        total = sum(data)
        value = total / n
        scaled = value * 100.0
        rounded_scaled = math.floor(scaled + 0.5)
        return rounded_scaled / 100.0

    def median(self, data):
        sorted_data = sorted(data)
        n = len(sorted_data)
        if n % 2 == 0:
            mid1 = sorted_data[n//2 - 1]
            mid2 = sorted_data[n//2]
            med = (mid1 + mid2) / 2.0
            scaled = med * 100.0
            rounded_scaled = math.floor(scaled + 0.5)
            return rounded_scaled / 100.0
        else:
            return float(sorted_data[n//2])

    def mode(self, data):
        if len(data) == 0:
            raise NoSuchElementException("No elements in data")
        
        freq = {}
        for num in data:
            freq[num] = freq.get(num, 0) + 1
        
        max_count = max(freq.values())
        modes = [key for key, count in freq.items() if count == max_count]
        modes.sort()
        return modes